### PR TITLE
core: tee: update objectSize/keySize for ECDSA/ECDH Objects

### PR DESCRIPTION
**Pull request by sahilnxp**
objectSize/keySize was not getting updated when an ECDSA/ECDH
object was imported.
Updating the ObjectSize/keySize based on the EC Curve.

Fixes: https://github.com/OP-TEE/optee_os/issues/2386
Signed-off-by: Sahil Malhotra <sahil.malhotra@nxp.com>
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>

### CODE REVIEW

File: core/tee/tee_svc_cryp.c [hunk 1]
```diff
@@ -1464,8 +1490,21 @@ static TEE_Result tee_svc_cryp_obj_populate_type(
 		 * of the object
 		 */
 		if (type_props->type_attrs[idx].flags &
-		    TEE_TYPE_ATTR_SIZE_INDICATOR)
+		    TEE_TYPE_ATTR_SIZE_INDICATOR) {
+		    /*
+		     * For ECDSA/ECDH we need to translate curve into
+		     * object size
+		     */
+		    if ((o->info.objectType == TEE_TYPE_ECDH_KEYPAIR)
```
Comment (jenswi-linaro) [inline core/tee/tee_svc_cryp.c]: Please test `attrs[n].attributeID == TEE_ATTR_ECC_CURVE` instead

**Comment by sahilnxp** [comment]
I have updated the pull request. Please check.

File: core/tee/tee_svc_cryp.c [hunk 2]
```diff
@@ -1464,8 +1489,19 @@ static TEE_Result tee_svc_cryp_obj_populate_type(
 		 * of the object
 		 */
 		if (type_props->type_attrs[idx].flags &
-		    TEE_TYPE_ATTR_SIZE_INDICATOR)
-			obj_size += attrs[n].content.ref.length * 8;
+			TEE_TYPE_ATTR_SIZE_INDICATOR) {
+		    /*
+		     * For ECDSA/ECDH we need to translate curve into
+		     * object size
+		     */
```
Comment (etienne-lms) [inline core/tee/tee_svc_cryp.c]: minor: alignement issue (use tabs)

File: core/tee/tee_svc_cryp.c [hunk 3]
```diff
@@ -1464,8 +1489,19 @@ static TEE_Result tee_svc_cryp_obj_populate_type(
...
+			if (attrs[n].attributeID == TEE_ATTR_ECC_CURVE) {
+				res = get_ec_key_size(attrs[n].content.value.a,
+					&obj_size);
+				if (res != TEE_SUCCESS)
+					return res;
+			} else
+				obj_size += attrs[n].content.ref.length * 8;
```
Comment (etienne-lms) [inline core/tee/tee_svc_cryp.c]: minor: add braces?

File: core/tee/tee_svc_cryp.c [hunk 4]
```diff
@@ -1464,8 +1489,19 @@ static TEE_Result tee_svc_cryp_obj_populate_type(
...
-		    TEE_TYPE_ATTR_SIZE_INDICATOR)
-			obj_size += attrs[n].content.ref.length * 8;
+			TEE_TYPE_ATTR_SIZE_INDICATOR) {
+		    /*
+		     * For ECDSA/ECDH we need to translate curve into
+		     * object size
+		     */
+			if (attrs[n].attributeID == TEE_ATTR_ECC_CURVE) {
+				res = get_ec_key_size(attrs[n].content.value.a,
+					&obj_size);
```
Comment (etienne-lms) [inline core/tee/tee_svc_cryp.c]: previous implementation increased the `obj_size` while this change initializes it.
(minor: maybe add few tabulations to align function argument)

**Comment by sahilnxp** [comment]
Pull request updated, Please check.

File: core/tee/tee_svc_cryp.c [hunk 5]
```diff
@@ -1464,8 +1489,19 @@ static TEE_Result tee_svc_cryp_obj_populate_type(
 		 * of the object
 		 */
 		if (type_props->type_attrs[idx].flags &
-		    TEE_TYPE_ATTR_SIZE_INDICATOR)
-			obj_size += attrs[n].content.ref.length * 8;
+			TEE_TYPE_ATTR_SIZE_INDICATOR) {
```
Comment (jenswi-linaro) [inline core/tee/tee_svc_cryp.c]: Please keep the indentation of this line.

File: core/tee/tee_svc_cryp.c [hunk 6]
```diff
@@ -1464,8 +1489,19 @@ static TEE_Result tee_svc_cryp_obj_populate_type(
...
+			/*
+			 * For ECDSA/ECDH we need to translate curve into
+			 * object size
+			 */
+			if (attrs[n].attributeID == TEE_ATTR_ECC_CURVE) {
+				res = get_ec_key_size(attrs[n].content.value.a,
+						      &obj_size);
+				if (res != TEE_SUCCESS)
+					return res;
+			} else
```
Comment (jenswi-linaro) [inline core/tee/tee_svc_cryp.c]: Please add braces after the `else` since there braces before the `else.

**Comment by sahilnxp** [comment]
Pushed again.
Sorry for again and again introducing the indentation issues.
One query, do optee-os follow kernel coding guidelines or something optee-os specific.
Will take care of these things next time. :)

**Comment by sahilnxp** [comment]
@jforissier , @etienne-lms please check, if there is any other comment.

**Comment by sahilnxp** [comment]
Jens tag applied.

**Comment by jforissier** [comment]
@sahilnxp thanks.

**Pull request merged**